external-crates/move/crates/move-vm-runtime/src/interpreter/mod.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{
+    interpreter::state::{CallFrame, MachineState},
+    loader::{arena::ArenaPointer, ast::Function, Loader},
+};
+use move_binary_format::errors::*;
+use move_vm_profiler::{profile_close_frame, profile_open_frame};
+use move_vm_types::{
+    data_store::DataStore, gas::GasMeter, loaded_data::runtime_types::Type, values::Value,
+};
+
+use crate::native_extensions::NativeContextExtensions;
+
+mod eval;
+pub(crate) mod state;
+
+/// Entrypoint into the interpreter. All external calls need to be routed through this
+/// function.
+pub(crate) fn run(
+    function: ArenaPointer<Function>,
+    ty_args: Vec<Type>,
+    args: Vec<Value>,
+    data_store: &impl DataStore,
+    gas_meter: &mut impl GasMeter,
+    extensions: &mut NativeContextExtensions,
+    loader: &Loader,
+) -> VMResult<Vec<Value>> {
+    let runtime_limits_config = &loader.vm_config().runtime_limits_config;
+    // TODO: Why does the VM config live on the loader?
+    let fun_ref = function.to_ref();
+    profile_open_frame!(gas_meter, fun_ref.pretty_string());
+
+    if fun_ref.is_native() {
+        let link_context = data_store.link_context();
+        let resolver = fun_ref.get_resolver(link_context, loader);
+
+        let return_values = eval::call_native_with_args(
+            None,
```

Comment:
Does it make sense for this to be None? Especially since it holds `VMRuntimeLimitsConfig`. Should we instead make an empty state of some sort?

Reply:
We could pass the limits separately. The only place this was used is in printing the stack trace.